attachment_preview/static/src/js/models/attachment_card/attachment_card.esm.js

```diff
@@ -0,0 +1,276 @@
+                                url: attachments[id].url,
+                                extension: extensions[id],
+                                title: attachments[id].title,
+                                previewUrl: getUrl(
+                                    id,
+                                    attachments[id].url,
+                                    extensions[id],
+                                    attachments[id].title
+                                ),
+                            };
+                        }
+                    );
+                    chatterpreviewableAttachments = reviewableAttachments;
+                });
+            },
+
+            /**
+             * @private
+             * @param {attachment_id} attachment_id of the attachment
+             * @param {attachment_url} attachment_url of the attachment
+             * @param {attachment_extension} attachment_extension of the attachment
+             * @param {attachment_title} attachment_title of the attachment
+             * @param {split_screen} split_screen of the attachment
+             */
+            _showPreview(
+                attachment_id,
+                attachment_url,
+                attachment_extension,
+                attachment_title,
+                split_screen
+            ) {
+                // Let active_attURL = "";
+                // this.attachmentList.attachments.forEach((att) => {
+                //     if (
+                //         parseInt(att.localId.slice(20).slice(0, -1)) === attachment_id
+                //     ) {
+                //         if (att.__values.url === undefined) {
+                //             att.__values.url = attachment_url.slice(
+                //                 window.location.origin.length
+                //             );
+                //             active_attURL = att.__values.url;
+                //         }
+                //     }
+                // });
+                var url = getUrl(
+                    attachment_id,
+                    attachment_url,
+                    attachment_extension,
+                    attachment_title
+                );
+
+                if (split_screen) {
+                    this.component.trigger("onAttachmentPreview", {
+                        url: url,
+                        active_attachment_id: active_attachment_id,
+                    });
+                } else {
+                    window.open(url);
+                }
+            },
+
+            /**
+             * @private
+             * @param {event} event
+             */
+            _onPreviewAttachment(event) {
+                event.preventDefault();
+
+                var self = this,
+                    $target = $(event.currentTarget),
+                    split_screen = $target.attr("data-target") !== "new",
+                    attachment_id = this.attachment.id,
+                    attachment_extension = "pdf",
```

Comment:
This should be removed and a get_attachment_extension rpc call should be used instead.

Reply:
@HolgerNahrstaedt we still need the variable `split_screen, attachment_id`  as the input for the rpc request. But you are right with the `attachment_extension`. It is unnecessary here, so I removed it per your suggestion.